lib/resty/core/log.lua

```diff
@@ -0,0 +1,98 @@
+-- Copyright (C) Yichun Zhang (agentzh)
```

Comment:
We should introduce a separate `ngx.log` module. The whole point of introducing a new separate Lua module is to stop adding more stuff into the global `ngx` table. But this module still adds more things to the `ngx` global.